fondant/dataset.py

```diff
@@ -166,82 +167,123 @@ def upload(self, save_path):
-                args=args,
-            )
-            # TODO update index, potentially add new subsets
+        component_arguments = {}
+        kubeflow_component_spec = self.spec.kubeflow_specification
+        component_arguments.update(kubeflow_component_spec.input_arguments)
+        component_arguments.update(kubeflow_component_spec.output_arguments)
+        return component_arguments
 
-        # step 4: create output manifest
-        output_manifest = dataset.upload(save_path=args.output_manifest_path)
+    @abstractmethod
+    def run(self):
+        """Abstract method for running component"""
 
-        return output_manifest
 
-    def _add_and_parse_args(self):
-        """
-        Add and parse component arguments based on the component specification.
-        """
+class FondantLoadComponent(FondantComponent):
+    """Abstract base class for a Fondant loader component"""
+
+    def _add_and_parse_args(self) -> argparse.Namespace:
         parser = argparse.ArgumentParser()
+        component_arguments = self.get_component_arguments()
 
-        kubeflow_component = self.spec.kubeflow_specification
+        for arg in component_arguments.values():
+            # Input manifest is not required for loading component
+            if arg.name == "input_manifest_path":
+                input_required = False
+            else:
+                input_required = True
 
-        # add input args
-        for arg in kubeflow_component.input_arguments.values():
             parser.add_argument(
                 f"--{arg.name}",
                 type=kubeflow2python_type[arg.type],
-                required=False
-                if self.type == "load" and arg.name == "input_manifest_path"
-                else True,
+                required=input_required,
                 help=arg.description,
             )
-        # add output args
-        for arg in kubeflow_component.output_arguments.values():
+
+        # add metadata
+        parser.add_argument(
+            "--metadata",
+            type=str,
+            required=True,
+            help="The metadata associated with the pipeline run",
+        )
+
+        return parser.parse_args()
+
+    def _get_manifest(self, args: argparse.Namespace) -> Manifest:
+        metadata = json.loads(args.metadata)
+        # TODO ideally get rid of args.metadata by including them in the storage args, getting
+        #  run_id based on args.output_manifest_path
+        manifest = Manifest.create(
+            base_path=metadata["base_path"],
+            run_id=metadata["run_id"],
+            component_id=metadata["component_id"],
+        )
+
+        return manifest
+
+    @abstractmethod
+    def load(self, args: argparse.Namespace) -> dd.DataFrame:
+        """Abstract method that loads the initial dataframe"""
+
+    def run(self):
+        """
+        Runs the loading component.
+        """
+        # step 1: Add and parse arguments
+        args = self._add_and_parse_args()
```

Comment:
Can we lift these common steps to the base class?

Reply:
Done :)